### PR TITLE
EDGECLOUD-3337: Unity: App Independent way to generate a Unique ID

### PR DESCRIPTION
**Pull request by lgarner**
Unity side of things. Android and IOS. Android has a separate implementation that is asynchronous via MobiledgeXAgent game object. It is not guaranteed it'll get a value before the rest of the app tries to use the value.

Requires C# SDK also updated to 2.1.3.

### CODE REVIEW

File: Runtime/Scripts/MobiledgeXAgent.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    {
+      lock (this)
+      {
+        NotifyHashAvertisingIdCallers -= request;
+      }
+    }
+
+    IEnumerator GetHashedAdvertisingID()
+    {
+      string hashedHexAdId = null;
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXAgent.cs:126]: Are you using hashedHexAdId somewhere ?

File: Runtime/Scripts/MobiledgeXAgent.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    public const int DEFAULT_ADVERTISING_ID_TIMEOUT_MS = 3000;
+    public int HashedAdvertisingIdTimeoutMs { get; set; } = DEFAULT_ADVERTISING_ID_TIMEOUT_MS;
+
+    private bool hashAdvertisingIdRequestedPrivate = false;
+    private bool hashAdvertisingIdRequested
+    {
+      get
+      {
+        string tmp = null;
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXAgent.cs:48]: tmp is not used ?!

File: Runtime/Scripts/MobiledgeXAgent.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      yield return StartCoroutine(GetHashedAdvertisingID());
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+      if (hashAdvertisingIdRequested)
+      {
+        //Debug.Log("ZZZ: Starting AdID coroutine:");
+        StartCoroutine(GetHashedAdvertisingID());
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXAgent.cs:76]: I don't know if you need to call **GetHashedAdvertisingID** in Update also
Although you have a lock its being called two times
Screen Shot 2020-08-03 at 11 27 44 AM

File: Runtime/Scripts/MobiledgeXIntegration.cs [hunk 4]
```diff
@@ -93,6 +96,10 @@ public MobiledgeXIntegration()
             matchingEngine = new MatchingEngine(pIntegration.CarrierInfo, pIntegration.NetInterface, pIntegration.UniqueID);
             melMessaging = new MelMessaging(appName);
             matchingEngine.SetMelMessaging(melMessaging);
+
+            Debug.Log("ZZZ: Manufacturer: " + melMessaging.GetManufacturer());
+            Debug.Log("ZZZ: Device Type: " + pIntegration.UniqueID.GetUniqueIDType());
+            Debug.Log("ZZZ: Device Ad ID: " + pIntegration.UniqueID.GetUniqueID());
```
Comment (franklin-huang-mobiledgex) [inline Runtime/Scripts/MobiledgeXIntegration.cs:102]: ZZZ?

Comment (lgarner) [reply Runtime/Scripts/MobiledgeXIntegration.cs]: ^ Will fix, we had a change of plans...

File: Runtime/Scripts/MobiledgeXAgent.cs (hunk 2 above)
Comment (lgarner) [reply Runtime/Scripts/MobiledgeXAgent.cs]: Impl now changed. Not needed currently

File: Runtime/Scripts/MobiledgeXAgent.cs (hunk 3 above)
Comment (lgarner) [reply Runtime/Scripts/MobiledgeXAgent.cs]: Removed function.

File: Runtime/Scripts/MobiledgeXAgent.cs (hunk 1 above)
Comment (lgarner) [reply Runtime/Scripts/MobiledgeXAgent.cs]: Removed. Stale.

**Comment by lgarner** [comment]
Simplified for now, but largely the same. 2.1.3 C# SDK update required still.

**Pull request merged**